GliaWidgets/Sources/ViewModel/Call/CallViewModel+CameraFlip.swift

```diff
@@ -0,0 +1,48 @@
+import Foundation
+
+extension CallViewModel {
+    func showCameraFlipIfNeeded() {
+        let flipCameraCallback: Cmd?
+
+        defer {
+            self.action?(.setCameraFlip(flipCameraCallback))
+        }
+
+        do {
+            let cameraDeviceManager = try environment.cameraDeviceManager()
+            let devices = cameraDeviceManager.cameraDevices().filter { $0.facing == .front || $0.facing == .back }
+            // Camera flip only makes sense when there are at least two camera devices.
+            guard devices.count >= 2 else {
+                flipCameraCallback = nil
+                return
+            }
+
+            flipCameraCallback = Cmd { [environment] in
+                // Get currently active camera device.
+                if let currentDevice = cameraDeviceManager.currentCameraDevice(),
+                    // Find corresponding index for active camera
+                    let currentIndex = devices.firstIndex(of: currentDevice) {
+                    // and increase it by one.
+                    let nextIndex = currentIndex.advanced(by: 1)
+                    // If next index is valid use it for next device selection, otherwise use first device.
+                    let nextDevice = devices.indices.contains(nextIndex) ? devices[nextIndex] : devices[0]
```

Comment:
Cool, this would mean that if there's a third camera at some point, this functionality wouldn't need changes, correct? Although then the name "flip" wouldn't be correct.

Reply:
Yes, still for now there are only `front` and `back` facing devices that are taken into account. Technically if two or more cameras will have only `front` facing,  toggling will also happen between them. But to be consistent with feature name and android implementation I think we can leave 'flip' name for now.